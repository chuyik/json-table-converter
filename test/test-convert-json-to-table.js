const { jsonToTableHtmlString } = require('../src')

const json = [
  {
    cacheKey: 'API_CACHE',
    report: {
      total: 2,
      topRecords: [
        {
          count: 1,
          uid: 'id-000000000000000001',
          username: 'chu',
        },
        {
          count: 2,
          uid: 'id-000000000000000002',
          username: 'yik',
        },
        'not-an-object',
      ],
    },
    date: '2018-07-30',
  },
]

describe('jsonToTableHtmlString()', () => {
  
  it('should convert json to table', () => {
    expect(jsonToTableHtmlString(json)).toMatchSnapshot()
  })

  })
})
