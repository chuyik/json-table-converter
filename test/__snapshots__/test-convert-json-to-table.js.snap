// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jsonToTableHtmlString() should convert json to table 1`] = `
"<table cellspacing=\\"0\\" style=\\"border-spacing: 0 0; border-color: #808080; border-collapse: collapse;\\">
  <thead><th style=\\"background: #F6F4F0;border: 1px solid #2d2d2d; padding: 3px;\\">cacheKey</th><th style=\\"background: #F6F4F0;border: 1px solid #2d2d2d; padding: 3px;\\">report</th><th style=\\"background: #F6F4F0;border: 1px solid #2d2d2d; padding: 3px;\\">date</th></thead>
  <tbody><tr style=\\"\\"><td style=\\"border: 1px solid #2d2d2d; padding: 3px;\\">API_CACHE</td><td style=\\"border: 1px solid #2d2d2d; padding: 3px;\\"><table cellspacing=\\"0\\" style=\\"width: 100%;border-spacing: 0 0; border-color: #808080; border-collapse: collapse;\\">
  <thead></thead>
  <tbody><tr style=\\"\\"><td style=\\"background: #F6F4F0; border: 1px solid #2d2d2d; padding: 3px;border: 1px solid #2d2d2d; padding: 3px;\\">total</td><td style=\\"border: 1px solid #2d2d2d; padding: 3px;\\">2</td></tr><tr style=\\"\\"><td style=\\"background: #F6F4F0; border: 1px solid #2d2d2d; padding: 3px;border: 1px solid #2d2d2d; padding: 3px;\\">topRecords</td><td style=\\"border: 1px solid #2d2d2d; padding: 3px;\\"><table cellspacing=\\"0\\" style=\\"width: 100%;border-spacing: 0 0; border-color: #808080; border-collapse: collapse;\\">
  <thead><th style=\\"background: #F6F4F0;border: 1px solid #2d2d2d; padding: 3px;\\">count</th><th style=\\"background: #F6F4F0;border: 1px solid #2d2d2d; padding: 3px;\\">uid</th><th style=\\"background: #F6F4F0;border: 1px solid #2d2d2d; padding: 3px;\\">username</th></thead>
  <tbody><tr style=\\"\\"><td style=\\"border: 1px solid #2d2d2d; padding: 3px;\\">1</td><td style=\\"border: 1px solid #2d2d2d; padding: 3px;\\">id-000000000000000001</td><td style=\\"border: 1px solid #2d2d2d; padding: 3px;\\">chu</td></tr><tr style=\\"\\"><td style=\\"border: 1px solid #2d2d2d; padding: 3px;\\">2</td><td style=\\"border: 1px solid #2d2d2d; padding: 3px;\\">id-000000000000000002</td><td style=\\"border: 1px solid #2d2d2d; padding: 3px;\\">yik</td></tr><tr style=\\"\\"><td style=\\"border: 1px solid #2d2d2d; padding: 3px;\\"></td><td style=\\"border: 1px solid #2d2d2d; padding: 3px;\\"></td><td style=\\"border: 1px solid #2d2d2d; padding: 3px;\\"></td><td style=\\"border: 1px solid #2d2d2d; padding: 3px;\\">not-an-object</td></tr></tbody>
</table></td></tr></tbody>
</table></td><td style=\\"border: 1px solid #2d2d2d; padding: 3px;\\">2018-07-30</td></tr></tbody>
</table>"
`;

exports[`jsonToTableHtmlString() should format cell 1`] = `
"<table cellspacing=\\"0\\" style=\\"border-spacing: 0 0; border-color: #808080; border-collapse: collapse;\\">
  <thead><th style=\\"background: #F6F4F0;border: 1px solid #2d2d2d; padding: 3px;\\">cacheKey</th><th style=\\"background: #F6F4F0;border: 1px solid #2d2d2d; padding: 3px;\\">report</th><th style=\\"background: #F6F4F0;border: 1px solid #2d2d2d; padding: 3px;\\">date</th></thead>
  <tbody><tr style=\\"\\"><td style=\\"border: 1px solid #2d2d2d; padding: 3px;\\">API_CACHE</td><td style=\\"border: 1px solid #2d2d2d; padding: 3px;\\"><table cellspacing=\\"0\\" style=\\"width: 100%;border-spacing: 0 0; border-color: #808080; border-collapse: collapse;\\">
  <thead></thead>
  <tbody><tr style=\\"\\"><td style=\\"background: #F6F4F0; border: 1px solid #2d2d2d; padding: 3px;border: 1px solid #2d2d2d; padding: 3px;\\">TOTAL</td><td style=\\"border: 1px solid #2d2d2d; padding: 3px;\\">2</td></tr><tr style=\\"\\"><td style=\\"background: #F6F4F0; border: 1px solid #2d2d2d; padding: 3px;border: 1px solid #2d2d2d; padding: 3px;\\">TOPRECORDS</td><td style=\\"border: 1px solid #2d2d2d; padding: 3px;\\"><table cellspacing=\\"0\\" style=\\"width: 100%;border-spacing: 0 0; border-color: #808080; border-collapse: collapse;\\">
  <thead><th style=\\"background: #F6F4F0;border: 1px solid #2d2d2d; padding: 3px;\\">count</th><th style=\\"background: #F6F4F0;border: 1px solid #2d2d2d; padding: 3px;\\">uid</th><th style=\\"background: #F6F4F0;border: 1px solid #2d2d2d; padding: 3px;\\">username</th></thead>
  <tbody><tr style=\\"\\"><td style=\\"border: 1px solid #2d2d2d; padding: 3px;\\">1</td><td style=\\"border: 1px solid #2d2d2d; padding: 3px;\\">id-000000000000000001</td><td style=\\"border: 1px solid #2d2d2d; padding: 3px;\\">chu</td></tr><tr style=\\"\\"><td style=\\"border: 1px solid #2d2d2d; padding: 3px;\\">2</td><td style=\\"border: 1px solid #2d2d2d; padding: 3px;\\">id-000000000000000002</td><td style=\\"border: 1px solid #2d2d2d; padding: 3px;\\">yik</td></tr><tr style=\\"\\"><td style=\\"border: 1px solid #2d2d2d; padding: 3px;\\"></td><td style=\\"border: 1px solid #2d2d2d; padding: 3px;\\"></td><td style=\\"border: 1px solid #2d2d2d; padding: 3px;\\"></td><td style=\\"border: 1px solid #2d2d2d; padding: 3px;\\">not-an-object</td></tr></tbody>
</table></td></tr></tbody>
</table></td><td style=\\"border: 1px solid #2d2d2d; padding: 3px;\\">2018-07-30</td></tr></tbody>
</table>"
`;
